tools/toolchains/xcode_configure/xcode_sdk_frameworks.bzl

```diff
@@ -0,0 +1,141 @@
+        ],
+    )
+    repository_ctx.delete(IMPORTS_FILE)
+    if deps_result.return_code != 0:
+        fail("Could not scan dependencies for {}\n{}".format(sdk_path, deps_result.stderr))
+    return json.decode(deps_result.stdout)
+
+C_MODULE_TMPL = """swift_c_module(
+    name = "{name}_c",
+    module_name = "{name}",
+    system_module_map = "{module_map_file}",
+    deps = [
+{deps}
+    ],
+)
+
+"""
+
+SWIFT_MODULE_TMPL = """swift_module_alias(
+    name = "{name}_swift",
+    deps = [
+{deps}
+    ]
+)
+
+"""
+
+def _sub_bazel_path_vars(path, sdk_path, developer_dir):
+    path = path.replace(str(sdk_path), "__BAZEL_XCODE_SDKROOT__")
+    path = path.replace(str(developer_dir), "__BAZEL_XCODE_DEVELOPER_DIR__")
+    return path
+
+def _rule_for_pkg(developer_dir, sdk_path, pkg):
+    module_path = pkg.get("modulePath", "")
+    if not module_path or IMPORTS_FILE in module_path:
+        return ""
+    module_name = module_path.replace(".pcm", "").replace(".swiftmodule", "")
+
+    pkg_deps = pkg.get("directDependencies", [])
+    deps = ["{}_c".format(d["clang"]) for d in pkg_deps if "clang" in d]
+    deps += ["{}_swift".format(d["swift"]) for d in pkg_deps if "swift" in d]
+    deps_string = "\n".join(sorted(["        \":{}\",".format(d) for d in deps]))
+
+    clang = pkg.get("details", {}).get("clang", {})
+    if clang:
+        return C_MODULE_TMPL.format(
+            name = module_name,
+            module_map_file = _sub_bazel_path_vars(clang["moduleMapPath"], sdk_path, developer_dir),
+            deps = deps_string,
+        )
+
+    return SWIFT_MODULE_TMPL.format(
+        name = module_name,
+        deps = deps_string,
+    )
+
+def _create_build_file_for_sdk(repository_ctx, developer_dir, sdk_path, output_folder, target_triple):
+    scan_deps = _get_dependency_graph(repository_ctx, developer_dir, sdk_path, target_triple)
+
+    build_file = _build_file_header()
+
+    for pkg in scan_deps.get("modules", []):
+        build_file += _rule_for_pkg(developer_dir, sdk_path, pkg)
+
+    build_file_path = output_folder.get_child("BUILD.bazel")
+    repository_ctx.file(build_file_path, build_file)
+
+def _platform_target_triple(platform, target_name, version):
+    return "{cpu}-apple-{platform}{version}{suffix}".format(
+        cpu = "arm64_32" if target_name == "watchos" else "arm64",
+        platform = target_name,
+        version = version,
+        suffix = "-simulator" if platform.endswith("Simulator") else "",
+    )
+
+def create_xcode_framework_targets(repository_ctx, xcode_version_name, developer_dir, versions):
+    """Creates a BUILD file for all the SDK frameworks contained in a given Xcode version."""
+    developer_dir_path = repository_ctx.path(developer_dir)
+    platforms_dir = developer_dir_path.get_child("Platforms")
+    for platform_name, target_name in PLATFORM_TUPLES:
+        platform_dir = platforms_dir.get_child(platform_name + ".platform")
+        platform_version = versions[target_name]
```

Comment:
For `Xcode 14` - it looks like if a `PLATFORM` on disk e.g. `tvos` - this will throw an error - potentially as a stopgap to make it work for iOS to buy us more time to fix it:
```
if not platform_dir.exists():
   print("Warning: skipping SDK for Xcode 14.0")
   continue
```

Reply:
I think the optional installs might just be for simulators? I don't have either of the optional platforms installed on my machine, but it generates the BUILD files just fine 🤔